### PR TITLE
fix unbalanced powerflow with out of service ext grids

### PR DESCRIPTION
**Pull request by jkupka**
Solves a mismatch of arrays during zero impedance addition of ext grids if at least one ext grid is out of service

### CODE REVIEW

File: pandapower/pd2ppc_zero.py [hunk 1]
```diff
@@ -421,11 +422,11 @@ def _add_ext_grid_sc_impedance_zero(net, ppc):
 
     # ext_grid zero sequence impedance
     if case == "max":
-        x0_grid = net.ext_grid["x0x_%s" % case].values * x_grid
-        r0_grid = net.ext_grid["r0x0_%s" % case].values * x0_grid
+        x0_grid = net.ext_grid[is_egs]["x0x_%s" % case].values * x_grid
```
Comment (dlohmeier) [inline pandapower/pd2ppc_zero.py:425]: To me ll. 425/426 and 428/429 look exactly the same. Could we just delete one part of both?

Comment (rbolgaryn) [reply pandapower/pd2ppc_zero.py:425]: I also see it this way

**Comment by vogt31337** [comment]
@jkupka @dlohmeier still relevant?

**Comment by jkupka** [comment]
yes, it's still relevant as it solves a different issue than #2184

**Review by rbolgaryn — CHANGES_REQUESTED** [review]
please address the comment from Daniel and add information to CHANGELOG.
Please also write a test that fails with the current pandapower and passes because with  this pull request.

**Comment by rbolgaryn** [comment]
@jkupka please address the change request

**Comment by jkupka** [comment]
The merge conflict was resolved. Failing tests seem to have nothing to do with the changes in this pull request

**Review by SteffenMeinecke — APPROVED** [review]

**Comment by SteffenMeinecke** [comment]
> The merge conflict was resolved. Failing tests seem to have nothing to do with the changes in this pull request

For some reason, I still cannot merge. Maybe `git rebase` is needed or something else. I don't know. Alternatively, I open another PR with the same fix.

**Comment by dlohmeier** [comment]
@rbolgaryn can you approve? Seems like it is not possible to merge without your approval.

**Review by dlohmeier — APPROVED** [review]

**Pull request merged**